sympy/functions/special/error_functions.py

```diff
@@ -192,14 +193,18 @@ def _eval_is_real(self):
         return self.args[0].is_extended_real
 
     def _eval_is_finite(self):
-        if self.args[0].is_finite:
-            return True
-        else:
-            return self.args[0].is_extended_real
+        z = self.args[0]
+        return fuzzy_or([z.is_finite, z.is_extended_real])
 
     def _eval_is_zero(self):
         return self.args[0].is_zero
 
+    def _eval_is_positive(self):
+        return self.args[0].is_extended_positive
+
+    def _eval_is_negative(self):
+        return self.args[0].is_extended_negative
```

Comment:
It seems like erf can be real for some nonreal arguments:
```python
In [38]: x, y = symbols('x, y', positive=True)

In [39]: erf(x + I*y).is_real
Out[39]: False

In [40]: v = nsolve(im(erf(x + I*y)).subs(x, 1), y, 2.56)

In [41]: v
Out[41]: 2.57021711858485

In [42]: erf(x + I*y).subs({x:1, y:v}).n()
Out[42]: -57.2392116972625 + 6.11478453961741e-15⋅ⅈ
```

Reply:
I apologize. I didn't consider it enough. I realized that `is_zero` also doesn't take complex numbers into account. I'll create a pull request to fix it.